src/pages/cli/graphql/data-modeling.mdx

```diff
@@ -0,0 +1,496 @@
+
+```graphql
+type Post @model {
+  id: ID!
+  title: String!
+  content: String
+  tags: [Tag] @manyToMany(relationName: "PostTags")
+}
+
+type Tag @model {
+  id: ID!
+  label: String!
+  posts: [Post] @manyToMany(relationName: "PostTags")
+}
+```
+
+Under the hood, the `@manyToMany` directive will create a "join table" named after the `relationName` to facilitate the many-to-many relationship. This generates queries and mutations that allow you to retrieve the related Comment records from the source Post record and vice versa:
+
+```graphql
+mutation CreatePost {
+  createPost(input: {title: "Hello World!!"}) {
+    id
+    title
+    content
+    tags { # queries the "join table" PostTags
+      items {
+        tag { # related Tag records from Post
+          id
+          label
+          posts { # queries the "join table" PostTags
+            items {
+              post { # related Post records from Tag
+                id
+                title
+                content
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+## Assign default values for fields
+
+You can use the `@default` directive to specify a default value for optional [scalar type fields](https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html) such as `Int`, `String`, and more. 
+
+```graphql
+type Todo @model {
+  content: String @default(value: "My new Todo")
+}
+```
+
+If you create a new Todo and don't supply a `content` input, Amplify will ensure that `My new Todo` is auto populated as a value. 
+
+## Advanced
+
+### Rename generated queries and mutations
+
+You can override the names of any `@model`-generated GraphQL queries, mutations and subscriptions, or remove operations. Just supply the desired prefix for the queries, mutations, and subscriptions. To disable a certain operation, just assign their value to `null`
+
+```graphql
+type Todo @model(queries: { get: "queryFor" }, mutations: null, subscriptions: null) {
+  content: String!
+}
+```
+
+In the example above, you'll be able to run a `queryForTodo` query to get a single Todo element. Mutations (meaning create, update, and delete) as well as subscriptions are disabled for the Todo model above.
+
+### Customize creation and update timestamps
+`@model` directive automatically adds `createdAt` and `updatedAt` timestamps to each entities. The timestamp field names can be changed by passing timestamps attribute to the directive.
+
+```graphql
+type Todo @model(timestamps:{createdAt: "createdOn", updatedAt: "updatedOn"}) {
+  content: String
+}
+```
+
+For example, the schema above will allow you to query for the following contents:
+
+```graphql
+type Todo {
+  id: ID!
+  content: String
+  createdOn: AWSDateTime!
+  updatedOn: AWSDateTime!
+}
+```
+
+### How it works
+The `@model` directive will generate:
```

Comment:
This section describes the `@model` directive, but there is no equivalent for `@primaryKey`, `@index`, `@default`, `@hasOne`, `@hasMany`, `@belongsTo`, or `@manyToMany`.